reframe/frontend/executors/policies.py

```diff
@@ -94,7 +94,7 @@ def runcase(self, case):
         self.stats.add_task(task)
         try:
             if (self.max_failures and
```

Comment:
In order to keep things simple, I suggest initializing `max_failures` to `sys.maxsize` and skip the first check here.